### PR TITLE
WIP: this probably fixes the bug. This is the fix from the C++ diff.

### PR DESCRIPTION
**Pull request by TeroFrondelius**
Trying to solve #19. 

@jw3126 can you review. This seems to fix the bug same way as in C++, not perfect but close.

### CODE REVIEW

File: test/test_broken.jl [hunk 1]
```diff
@@ -8,22 +8,22 @@ using Miniball
...
 
-    @test_throws ErrorException miniball(Float64.(pts))
+    #@test_throws ErrorException miniball(Float64.(pts))
     ball = miniball(Float64.(pts), check=false)
     ball_radius = sqrt(ball.squared_radius)
     pt = pts[1,:]
-    @test_broken norm(pt - ball.center) <= ball_radius * 1.4
-    @test_broken norm(pt - ball.center) <= ball_radius
+    @test norm(pt - ball.center) <= ball_radius * 1.4
```
Comment (jw3126) [inline test/test_broken.jl]: You can delete the test with the 1.4 was just to show that the error is that bad. Same for the 1.7 below.

**Comment by jw3126** [comment]
I tried this branch with my actual use case and unfortunately it produces still failures. (Also some of the random tests in CI still fail.)
OTOH the boundary of a ball is a fuzzy concept in finite precision arithmetic, so maybe it is not possible to design a stable algorithm around it.

**Comment by jw3126** [comment]
Ok it seems the cxx version is much better then this, see #22.

**Comment by TeroFrondelius** [comment]
After @ahojukka5 refactoring this pull request is not meaningful anymore.

**Pull request closed without merging**